### ISSUE TITLE
Page unusable with "Delete cookies and site data when Firefox is closed" preference

### ISSUE THREAD

**Issue by aibaraiduas** [mozilla/send#1226]
If the "Delete cookies and site data when Firefox is closed" box is checked in Firefox's preferences (Privacy & Security -> Cookies and Site Data), the send page (send.firefox.com) fails to properly load and it unusable. Currently using Firefox 65.0.1.

![Screenshot from 2019-03-12 19-10-34](https://user-images.githubusercontent.com/5825551/54242614-a0fe2100-44fb-11e9-9131-e26dcbe0bd2e.png)

I've had that option ticked for as long as I can remember and send used to work fine. Now the network inspector gives this error:
`Failed to register/update a ServiceWorker for scope ‘https://send.firefox.com/’: Storage access is restricted in this context due to user settings or private browsing mode. main.js:38:10
SecurityError: The operation is insecure.`

**Comment by dannycoates** [comment]
Thanks for the report @aibaraiduas. The fix for this will go out in our next release. The download experience will be degraded, i.e. large files may not work as well, but the site should be usable.

**Comment by SoftVision-PaulOiegas** [comment]
hey @dannycoates, this issue intrigued me a bit because the option says "... when Firefox is closed". So if I have send loaded in one tab, and I check this option, after I refresh the page Send gets instantly broken. So checking the option changes something that instantly breaks Send, not only at browser close.

Found out what pref is modified by this option ("network.cookie.lifetimePolicy" gets value 2 from 0). Then  I've did a bit of digging with mozregression tool and found out that something changed in Firefox 58 that made this behavior occur. Last good build was 2017-09-20 and first bad 2017-09-21.
Here is the [pushlog](https://hg.mozilla.org/mozilla-central/pushloghtml?fromchange=319a34bea9e4f3459886b5b9e835bd338320f1fd&tochange=ca7d18dbacbf103d74a3213d8d08a7c3e4def9a2).
Here is a bug ([1336364](https://bugzilla.mozilla.org/show_bug.cgi?id=1336364)) that raised my attention from it.

Long story short, I was wondering if this issue is not caused by a Firefox issue/feature implementation. Probably you'd understand better the changes made in the bug and can figure out what exactly is happening.

**Comment by bobfe-ids** [comment]
That explains why I occasionally get messages from sites that I won't accept cookies. It's not that the browser actually stops accepting cookies, it changes the expiration value that gets written in a cookie to always be 'Session'. I agree with @SoftVision-PaulOiegas the problem seems to be in what that option does. It no longer does what it says, just "Delete" cookies without actually changing them.

I found that I was able to leave that option unchecked and by checking  "Clear history when firefox closes" -> "Settings" cookies and site data can still be set to be deleted.
